refactor(language_server): move `nested_configs` to `ServerLinter`

Sysix opened the pull request:
Normally we would just restart the `ServerLinter.isolated_linter` or the `Worker.nested_configs`. 
I do not see the benefits to store the `nested_configs` separately and changing them + the `isolated_linter` when needed.
And I do not want to change `isolated_linter` from `Arc` to `RwLock` / `Mutex`.

When a `.oxlintrc.json` file changes, we just set up the linter and searching for all nested configs again.

Sysix commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10775?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10775** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10775?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10775?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#10762** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10762?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#10761** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10761?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10775).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented on crates/oxc_language_server/src/linter/server_linter.rs:
```diff
@@ -24,9 +24,73 @@ pub struct ServerLinter {
+        let relative_config_path = options.config_path.clone();
+        let oxlintrc = if relative_config_path.is_some() {
+            let config = root_path.join(relative_config_path.unwrap());
+            if config.try_exists().expect("Could not get fs metadata for config") {
```
should thie be `.ok()` rather than panicing?

Sysix replied:
`ok()` is not enough, changed it to `if config.try_exists().is_ok_and(|exists| exists) {`

camc314 replied:
ah yes you're right

Sysix commented:
### Merge activity

* **May 7, 9:29 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
refactor(language_server): move `nested_configs` to `ServerLinter` (#10775)

Normally we would just restart the `ServerLinter.isolated_linter` or the `Worker.nested_configs`.
I do not see the benefits to store the `nested_configs` separately and changing them + the `isolated_linter` when needed.
And I do not want to change `isolated_linter` from `Arc` to `RwLock` / `Mutex`.

When a `.oxlintrc.json` file changes, we just set up the linter and searching for all nested configs again.
